linkerd/app/integration/src/controller.rs

```diff
@@ -327,40 +326,43 @@ where
     let (listening_tx, listening_rx) = oneshot::channel();
     let mut listening_tx = Some(listening_tx);
 
-    ::std::thread::Builder::new()
+    let (trace, _) = trace_init();
+    std::thread::Builder::new()
         .name(name.into())
         .spawn(move || {
-            if let Some(delay) = delay {
-                let _ = listening_tx.take().unwrap().send(());
-                delay.wait().expect("support server delay wait");
-            }
-            let mut runtime = runtime::current_thread::Runtime::new().expect("support runtime");
+            tracing::dispatcher::with_default(&trace, move || {
```

Comment:
we could also use `set_default` here, which returns a drop guard, and avoid indenting everything...it's the same change, but makes a smaller diff. I should've mentioned that in my earlier review.

Reply:
(not a blocker, just a note)